pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1018,6 +1018,22 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 	return nil
 }
 
+func calculateRequestedSockets(vmi *v1.VirtualMachineInstance) uint32 {
+	sockets := uint32(1)
+	if vmi.Spec.Domain.CPU != nil && vmi.Spec.Domain.CPU.Sockets != 0 {
+		return vmi.Spec.Domain.CPU.Sockets
+	}
+	return sockets
+}
+
+func calculateRequestedThreads(vmi *v1.VirtualMachineInstance) uint32 {
+	threads := uint32(1)
+	if vmi.Spec.Domain.CPU != nil && vmi.Spec.Domain.CPU.Threads != 0 {
+		return vmi.Spec.Domain.CPU.Threads
+	}
+	return threads
+}
+
```

Comment:
I think we need to make sure that sockets, threads, cores (and any combination thereof) falls under the constraints set by the pod. AFAIK/R, kubernetes only knows about (fractions of) cores, considering them like execution unit.
Here we are shuffling things around, meaning that, given for example 2 kubevirt "cores" (execution unit) we can either consider them
- 2 thread 1 core 1 socket CPU
- 1 thread 2 core 1 socket CPU
- 1 thread 1 core 2 socket CPU

things can get messy pretty fast (what about 16 kubevirt "cores" :)). I think we should at least laid foundations to ensure we adhere to the limits (eg. we can't somehow get to have 16 threads 16 cores 16 socket CPUs)